pandas/tests/frame/test_missing.py

```diff
@@ -270,6 +270,80 @@ def test_fillna(self):
                                   pd.Timestamp('2012-11-11 00:00:00+01:00')]})
         assert_frame_equal(df.fillna(method='bfill'), exp)
 
+    def test_na_actions(self):
+
+        cat = pd.Categorical([1, 2, 3, np.nan], categories=[1, 2, 3])
+        vals = ["a", "b", np.nan, "d"]
+        df = pd.DataFrame({"cats": cat, "vals": vals})
+        cat2 = pd.Categorical([1, 2, 3, 3], categories=[1, 2, 3])
+        vals2 = ["a", "b", "b", "d"]
+        df_exp_fill = pd.DataFrame({"cats": cat2, "vals": vals2})
+        cat3 = pd.Categorical([1, 2, 3], categories=[1, 2, 3])
+        vals3 = ["a", "b", np.nan]
+        df_exp_drop_cats = pd.DataFrame({"cats": cat3, "vals": vals3})
+        cat4 = pd.Categorical([1, 2], categories=[1, 2, 3])
+        vals4 = ["a", "b"]
+        df_exp_drop_all = pd.DataFrame({"cats": cat4, "vals": vals4})
+
+        # fillna
+        res = df.fillna(value={"cats": 3, "vals": "b"})
+        tm.assert_frame_equal(res, df_exp_fill)
+
+        def f():
+            df.fillna(value={"cats": 4, "vals": "c"})
+
```

Comment:
this test is way too long. can you parametrize it? would make it simpler. could also break it into a couple of tests, just use descriptive names.

Reply:
haha, realized you are just moving this test! ok see what you can do here.